### PR TITLE
Improved hierarchical traversal

### PR DESCRIPTION
**Pull request by mallport**
This PR adds the following:

* New module: `globberator` [Link](src/dapla_pseudo/globberator/traverser.py). This module takes in a Polars Schema, a set of `PseudoRules` with glob patterns, and generates new `PseudoRules` _with_ concrete paths.

* Rules with concrete paths are now used while traversing hierarchical datasets. This will (hopefully) cut down significantly on the time used for traversal, due to there being way more exit ramps for wrongly matched data.

* Split the hierarchical and non-hierarchical logic. Now, 2D datasets will not be converted to JSON and instead use the regular indexing you would typically use to select columns from a DataFrame. This should ease memory and CPU usage. From the user side, hierarchical datasets are indicated like so:

```python
result = (
        Pseudonymize.from_polars(df_personer_hierarchical)
        .add_rules(rule)
        .run(hierarchical=True)
    )
```

### CODE REVIEW

**Review by skykanin — APPROVED** [review]
We should update the dapla manual to reflect this change for the users

**Comment by kschulst** [comment]
Nice work! Lots of (well documented) code and stuff that is going on, so would be nice with a huddle walk through on this one.

File: src/dapla_pseudo/v1/mutable_dataframe.py [hunk 1]
```diff
@@ -45,55 +45,92 @@ def __init__(
         self.func = func
         self.target_func = target_func
 
-    def update_col(self, key: str, data: list[str]) -> None:
-        """Update the values in the matched column."""
-        self.col.update({key: data})
+    def get_value(self) -> list[str | int | None]:
+        """Get the inner value.
+
+        If hierarchical, get the values of the matched column.
```
Comment (kschulst) [inline src/dapla_pseudo/v1/mutable_dataframe.py:51]: Would be nice if we could deduce this somehow?

Maybe document what how this affects the processing

**Review by kschulst — APPROVED** [review]
Some comments, but would be happy to discuss in a walk-through instead, since I haven't tested the code on a real life case.

**Pull request merged**